controllers/provider-aws/pkg/apis/aws/v1alpha1/types_infrastructure.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+const (
+	// EventReasonDestruction an event describing infrastructure destruction
+	EventReasonDestruction string = "InfrastructureDestruction"
+	// EventReasonCreation an event describing infrastructure creation
+	EventReasonCreation string = "InfrastructureCreation"
+
+	// PurposeNodes is the purpose for nodes
+	PurposeNodes string = "nodes"
```

Comment:
This constant must also be defined in the internal version as the AWS controller will work with the internal version.